### PR TITLE
[10.0][ADD] sale_order_restrict_cancel_existing_invoice

### PR DESCRIPTION
**Pull request by rousseldenis**
Allows to restrict Sale Order cancellation if a confirmed invoice
exists

### CODE REVIEW

**Comment by rousseldenis** [comment]
@Cedric-Pigeon

**Review by acsonefho — APPROVED** [review]
LGTM (Code only)

File: sale_order_restrict_cancel_existing_invoice/README.rst [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+.. contents::
+   :local:
+
+Usage
+=====
+
+* Create a Sale Order and confirm it (and deliver products if needed)
+* Create Invoice and confirm it
+* On Sale Order, try to click on 'Cancel'. An error should occur
```
Comment (HaraldPanten) [inline sale_order_restrict_cancel_existing_invoice/README.rst:41]: Just a minor comment.

If you have a delivered picking from a SO, user wouldn't be able to cancel this Sale Order, anyway.

**Review by HaraldPanten — APPROVED** [review]
Functional review. LGTM

File: sale_order_restrict_cancel_existing_invoice/README.rst (hunk 1 above)
Comment (rousseldenis) [reply sale_order_restrict_cancel_existing_invoice/README.rst:41]: But if you haven't delivered nothing, this case applies.

Typically, that's the case of advance amount which is invoiced. Refuse to cancel such order. First, do a refund invoice (which will mark the advance invoice as paid). Then, you can cancel the sale.

Comment (HaraldPanten) [reply sale_order_restrict_cancel_existing_invoice/README.rst:41]: Maybe I didn't explain myself well enough. I just commented that if you follow the steps you're describing in the README file (delivering products) you'll not be able to cancel the sale order as you can not cancel it if you have delivered quantities. Anyway, this PR looks good to me 👍 . It was just a minor comment.

**Review by Cedric-Pigeon — APPROVED** [review]

**Pull request merged**